lib/bwa/message.rb

```diff
@@ -58,18 +58,28 @@ def parse(data)
           # Not enough data for a full message; return and hope for more
           return nil if data.length - offset < 5
 
+          # Keep scanning until message start char
           next unless data[offset] == '~'
+
+          # Read length (safe since we have at least 5 chars)
           length = data[offset + 1].ord
-          # impossible message
-          next if length < 5
+
+          # No message is this short or this long; keep scanning
+          next if length < 5 or length >= '~'.ord
 
           # don't have enough data for what this message wants;
-          # it could be garbage on the line so keep scanning
-          next if length + 2 > data.length - offset
+          # return and hope for more (yes this might cause a
```

Comment:
so yes, the serial protocol is _very_ chatty, but when connecting via the official wifi adapter it is much less so. still fairly chatty - about 1 message per second iirc. which is enough that I'm still comfortable with this change

Reply:
Ah, I guess I wasn't thinking about that. We could tweak the length limit to be the maximum recognized message length (currently 32) to improve matters. But if you're happy with it as-is, great! Thanks!